models/demos/tg/llama3_70b/tt/llama_decoder_galaxy.py

```diff
@@ -83,57 +82,65 @@ def __init__(
-        )
-
-        self.LN_PROGCFG = ttnn.LayerNormShardedMultiCoreProgramConfig(
-            compute_with_storage_grid_size=[8, 4],
-            subblock_w=8,
-            block_h=32 // 32,
-            block_w=8,
-            inplace=False,
-        )
-
-        shard_spec_32_cores_grid = ttnn.CoreRangeSet(
-            {
-                ttnn.CoreRange(
-                    ttnn.CoreCoord(0, 0),
-                    ttnn.CoreCoord(7, 3),
+        if self.model_config["LLM_MODE"] == "decode":
+            self.LN_COMPUTE_KERNEL_CONFIG = ttnn.WormholeComputeKernelConfig(
+                math_fidelity=ttnn.MathFidelity.HiFi2,
+                math_approx_mode=False,
+                fp32_dest_acc_en=False,
+                packer_l1_acc=False,
+            )
+
+            self.LN_PROGCFG = ttnn.LayerNormShardedMultiCoreProgramConfig(
+                compute_with_storage_grid_size=[8, 4],
+                subblock_w=8,
+                block_h=32 // 32,
+                block_w=8,
+                inplace=False,
+            )
+
+            shard_spec_32_cores_grid = ttnn.CoreRangeSet(
+                {
+                    ttnn.CoreRange(
+                        ttnn.CoreCoord(0, 0),
+                        ttnn.CoreCoord(7, 3),
+                    ),
+                }
+            )
+
+            self.LN_OUTPUT_MEMCFG = ttnn.MemoryConfig(
+                ttnn.TensorMemoryLayout.WIDTH_SHARDED,
+                ttnn.BufferType.L1,
+                ttnn.ShardSpec(
+                    shard_spec_32_cores_grid,
+                    [
+                        32,
+                        8192 // 32,
+                    ],
+                    ttnn.ShardOrientation.ROW_MAJOR,
+                    False,
                 ),
-            }
-        )
-
-        self.LN_OUTPUT_MEMCFG = ttnn.MemoryConfig(
-            ttnn.TensorMemoryLayout.WIDTH_SHARDED,
-            ttnn.BufferType.L1,
-            ttnn.ShardSpec(
-                shard_spec_32_cores_grid,
-                [
-                    32,
-                    8192 // 32,
-                ],
-                ttnn.ShardOrientation.ROW_MAJOR,
-                False,
-            ),
-        )
-        self.ATTN_ACT_MEMCFG = ttnn.create_sharded_memory_config(
-            shape=(32, 2048 // 32),
-            core_grid=ttnn.CoreGrid(y=4, x=8),
-            strategy=ttnn.ShardStrategy.WIDTH,
-            orientation=ttnn.ShardOrientation.ROW_MAJOR,
-            use_height_and_width_as_shard_shape=True,
-        )
-        self.MLP_ACT_MEMCFG = ttnn.create_sharded_memory_config(
-            shape=(32, 2048 // 8),
-            core_grid=ttnn.CoreGrid(y=1, x=8),
-            strategy=ttnn.ShardStrategy.WIDTH,
-            orientation=ttnn.ShardOrientation.ROW_MAJOR,
-            use_height_and_width_as_shard_shape=True,
-        )
+            )
+            self.ATTN_ACT_MEMCFG = ttnn.create_sharded_memory_config(
+                shape=(32, 2048 // 32),
```

Comment:
use model_config variables here, avoid any hardcoded numbers

Reply:
This is decode part of the code, I didn't change it, just put under if statement block so I am not sure whether the dims are (batch, hidden_dim // cluster_shape[0] // num_of_chips) or there are some other variables intended to be.